src/main/java/com/google/cloud/tools/managedcloudsdk/process/ProcessExecutor.java

```diff
@@ -73,7 +77,7 @@ public int run(
       exitCode = process.waitFor();
     } catch (InterruptedException ex) {
       process.destroy();
-      throw new InterruptedException("Process cancelled.");
+      throw ex; // rethrow after cleanup
```

Comment:
Can this just be like `finally { process.destroy(); }` (without the catch)?

Reply:
I think destroying the process before the streams have a chance to fully consume the output could lead to IOExceptions.

Reply:
for the case when the process has exited gracefully, and we still call "destroy"

Reply:
Okay that makes sense.